src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -50,6 +50,14 @@ public static class ExpressionMetadataProvider
                     // Property/field access is always legal
                     var memberExpression = (MemberExpression)expression.Body;
                     propertyName = memberExpression.Member is PropertyInfo ? memberExpression.Member.Name : null;
+                    if (string.Equals(propertyName, "Model", StringComparison.Ordinal) &&
```

Comment:
Is this though enough? Should we check the expression on the LHS? What about a case like:
```
public class Thing
{
    public Thing Model { get; set; }
}
```